bftengine/src/bftengine/messages/ViewChangeMsg.cpp

```diff
@@ -178,6 +216,32 @@ bool ViewChangeMsg::checkElements(uint16_t sigSize) const {
   return true;
 }
 
+bool ViewChangeMsg::checkComplaints(uint16_t sigSize) const {
+  uint16_t numOfActualComplaints = 0;
+  uint32_t remainingBytes = size() - sigSize - b()->locationAfterLast;
+  char* currLoc = body() + b()->locationAfterLast + sigSize;
+
+  while (remainingBytes > sizeof(MessageBase::Header) && (numOfActualComplaints < numberOfComplaints())) {
+    MsgSize* complaintSize = (MsgSize*)currLoc;
+    remainingBytes -= sizeof(MsgSize);
+    currLoc += sizeof(MsgSize);
+
+    if (*complaintSize <= sizeof(MessageBase::Header)) {
+      return false;
+    }
+
+    numOfActualComplaints++;
+    remainingBytes -= *complaintSize;
+    currLoc += *complaintSize;
+  }
+
+  if (numOfActualComplaints != numberOfComplaints()) return false;
+
+  if (remainingBytes != 0) return false;
+
+  return true;
```

Comment:
shouldn't we also check that we have at least `f+1` signatures?

Reply:
I think might be better to leave to higher level to verify the signatures and the quorum size.